src/crypto/ecma376_agile.js

```diff
@@ -1,493 +1,501 @@
+const zip = require('../util/zip');
+
+
+const ENCRYPTION_INFO_PREFIX = Buffer.from([0x04, 0x00, 0x04, 0x00, 0x40, 0x00, 0x00, 0x00]); // First 4 bytes are the version number, second 4 bytes are reserved.
+const PACKAGE_ENCRYPTION_CHUNK_SIZE = 4096;
+const PACKAGE_OFFSET = 8; // First 8 bytes are the size of the stream
+
+// Block keys used for encryption
+const BLOCK_KEYS = {
+  dataIntegrity: {
+    hmacKey: Buffer.from([0x5f, 0xb2, 0xad, 0x01, 0x0c, 0xb9, 0xe1, 0xf6]),
+    hmacValue: Buffer.from([0xa0, 0x67, 0x7f, 0x02, 0xb2, 0x2c, 0x84, 0x33]),
+  },
+  key: Buffer.from([0x14, 0x6e, 0x0b, 0xe7, 0xab, 0xac, 0xd0, 0xd6]),
+  verifierHash: {
+    input: Buffer.from([0xfe, 0xa7, 0xd2, 0x76, 0x3b, 0x4b, 0x9e, 0x79]),
+    value: Buffer.from([0xd7, 0xaa, 0x0f, 0x6d, 0x30, 0x61, 0x34, 0x4e]),
+  },
+};
+
+const Encryptor = {
+  /**
+   * Calculate a hash of the concatenated buffers with the given algorithm.
+   * @param {string} algorithm - The hash algorithm.
+   * @return {Buffer} The hash
+   */
+  hash(algorithm, ...buffers) {
+    const hash = crypto.createHash(algorithm);
+    hash.update(Buffer.concat(buffers));
+    return hash.digest();
+  },
+
+  /**
+   * Decrypt the data with the given password
+   * @param {Buffer} data - The data to decrypt
+   * @param {string} password - The password
+   * @return {Promise.<Buffer>} The decrypted data
+   */
+  async decrypt(data, password) {
+    let {encryptionInfoBuffer, encryptedPackageBuffer} = data;
+
+    // In the browser the CFB content is an array. Convert to a Buffer.
+    if (!Buffer.isBuffer(encryptionInfoBuffer)) encryptionInfoBuffer = Buffer.from(encryptionInfoBuffer);
+    if (!Buffer.isBuffer(encryptedPackageBuffer)) encryptedPackageBuffer = Buffer.from(encryptedPackageBuffer);
+
+
+    // Parse the encryption info XML into an object
+    const encryptionInfo = await this.parseEncryptionInfo(encryptionInfoBuffer);
+
+    // Convert the password into an encryption key
+    const key = this.convertPasswordToKey(
+        password,
+        encryptionInfo.key.hashAlgorithm,
+        encryptionInfo.key.saltValue,
+        encryptionInfo.key.spinCount,
+        encryptionInfo.key.keyBits,
+        BLOCK_KEYS.key,
+    );
+
+    // Use the key to decrypt the package key
+    const packageKey = this.crypt(
+        false,
+        encryptionInfo.key.cipherAlgorithm,
+        encryptionInfo.key.cipherChaining,
+        key,
+        encryptionInfo.key.saltValue,
+        encryptionInfo.key.encryptedKeyValue,
+    );
+
+    // Use the package key to decrypt the package
+    const outputFileBuffer = this.cryptPackage(
+        false,
+        encryptionInfo.package.cipherAlgorithm,
+        encryptionInfo.package.cipherChaining,
+        encryptionInfo.package.hashAlgorithm,
+        encryptionInfo.package.blockSize,
+        encryptionInfo.package.saltValue,
+        packageKey,
+        encryptedPackageBuffer,
+    );
+
+    const isValidZip = zip.isValidZip(outputFileBuffer);

```

Comment:
Changes are just here but the file is completely reformatted as mentioned, please help me with that.